sdk/core/Azure.Core/src/Shared/OperationHelpers.cs

```diff
@@ -53,5 +53,72 @@ public static async ValueTask<Response<TResult>> DefaultWaitForCompletionAsync<T
                 await Task.Delay(pollingInterval, cancellationToken).ConfigureAwait(false);
             }
         }
+
+        public static Response<TResult> DefaultWaitForCompletion<TResult>(this Operation<TResult> operation, CancellationToken cancellationToken)
+            where TResult : notnull
+        {
+            return operation.DefaultWaitForCompletion(DefaultPollingInterval, cancellationToken);
+        }
+
+        public static Response<TResult> DefaultWaitForCompletion<TResult>(
+            this Operation<TResult> operation,
+            TimeSpan pollingInterval,
+            CancellationToken cancellationToken)
+            where TResult : notnull
+        {
+            while (true)
+            {
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
+                operation.UpdateStatusAsync(cancellationToken).GetAwaiter().GetResult();
```

Comment:
Would it make more sense if we just called the sync `UpdateStatus` directly?

Reply:
Good catch!